src/interfaces/swig/SGBase.i

```diff
@@ -411,6 +411,18 @@ namespace shogun
             }
         }
 
+        bool parameter_is_sg_base(const std::string& name) const {
```

Comment:
maybe briefer: `is_sg_base(const std::string& param_name)`? Or is that confusing?

Reply:
actually, could we make this a function in shogun namespace, rather than a member?
Also, would it be possible to do that with templates then? :D

Reply:
not sure I was thinking straight when I wrote this but I had in mind an overloaded function that you pass something and it will give you a bool and we specialize the template for sg base types. However, I think with swig this will just give headaches, so maybe the current thing is best

Reply:
Ah ok! But what would the signature be? Can give it a go, this method is tailored for what I need rather than what a user would need, I think..

Reply:
Like a method with one argument and bool return. It takes a templated argument and returns false. And then it is spezialized for shogun base classes and then returns true. Not sure that would work :)
If it is only you who needs this the current approach is probably fine as well or?

Reply:
It might not be worth adding all those templates just for this? Can always change in the future if it turns out to be useful.

Reply:
yeah nevermind, I was just fantasizing :)